tests/test_resolver.py

```diff
@@ -73,6 +73,48 @@ def test_resolvable_set():
     rs.merge(rq, [binary_pkg])
 
 
+def test_resolvable_set_is_constraint_only():
+  builder = ResolverOptionsBuilder()
+  rs = _ResolvableSet()
+  c = ResolvableRequirement.from_string('foo', builder)
+  c.is_constraint = True
+
+  package = SourcePackage.from_href('foo-2.3.4.tar.gz')
+  rs.merge(c, [package])
+
+  assert rs.packages() == [(c, set([package]), None, True)]
+
+
+def test_resolvable_set_constraint_and_non_constraint():
+  builder = ResolverOptionsBuilder()
+  rs = _ResolvableSet()
+  constraint = ResolvableRequirement.from_string('foo', builder)
+  constraint.is_constraint = True
```

Comment:
yes, that makes sense. the latter sounds like a unit test that should already exist (since that's just testing standard requirement behavior), but I can add both.